File: library/src/main/java/ga/rugal/git/model/GitBlob.java
package ga.rugal.git.model;

import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
import lombok.EqualsAndHashCode;
import lombok.Value;

/**
 * Problematic file.
 *
 * @author Rugal Bernstein
 */
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@SuppressFBWarnings({"EI_EXPOSE_REP", "EI_EXPOSE_REP2"})
@Value
public class GitBlob {

  /**
   * SHA-1 of the git object: blob.
   */
  @EqualsAndHashCode.Include
  String id;

  /**
   * The file path in repository.
   */
  String path;

  /**
   * File size in file system.
   */
  long size;
}

File: library/src/main/kotlin/ga/rugal/gitcleaner/core/model/GitBlob.kt
package ga.rugal.gitcleaner.core.model

/**
 * Git blob mapping.
 *
 * @author Rugal Bernstein
 */
data class GitBlob(
  /**
   * SHA-1 of the git object: blob.
   */
  val id: String,
  /**
   * The file path in repository.
   */
  val path: String,
  /**
   * File size byte in file system.
   */
  val size: Long,
) {
  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (javaClass != other?.javaClass) return false

    other as GitBlob

    return id == other.id
  }

  override fun hashCode(): Int = id.hashCode()
}
